http-server/src/main/java/io/micronaut/servlet/http/server/HttpExchangeHttpServletResponse.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.servlet.http.server;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpStatus;
+import jakarta.servlet.ServletOutputStream;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+
+/**
+ * {@link HttpServletResponse} implementation backed with a {@link ByteArrayOutputStream}.
+ */
+@Experimental
+@Internal
+class HttpExchangeHttpServletResponse implements HttpServletResponse {
```

Comment:
again doesn't it make sense to implement Micronaut's interface instead of bridging servlet?